frontend/src/app/shared/components/modals/modal-wrapper/dynamic-content.modal.ts

```diff
@@ -63,7 +63,18 @@ export class DynamicContentModalComponent extends OpModalComponent implements On
 
     const modal = document.querySelector('.spot-modal') as HTMLElement;
     const closeButton = modal.querySelector('[dynamic-content-modal-close-button]') as HTMLMetaElement;
-    closeButton.addEventListener('click', () => this.closeMe());
+    if (closeButton != null) {
+      closeButton.addEventListener('click', () => this.closeMe());
+    }
+
+    const closeModalAndPrintElement = modal.querySelector('.op-export-options--option-print');
```

Comment:
This is not the correct place to put this logic. However, as the export modal is not an angular component, there is no direct place to put this currently so I can understand you putting it here for now.

Two alternative approaches:

1. Add a new callback to the global click listener in https://github.com/opf/openproject/blob/174fd81fcb1dfd58162ce3fb5c7d2ba172e1fabe/frontend/src/app/core/setup/globals/global-listeners.ts#L52-L57, checking for a specific identifier (I'm thinking about something like `data-link-action="print"` so we can easily grep for it. This link might be added to other export modals so would make sense to generalize it

2. Add a new dynamically loaded JS script that adds the event listener specifically on the wiki page

To do that, add a new file `frontend/src/app/core/augmenting/dynamic-scripts/wiki_export_modal.js`

and reference it with the following meta tag
```
<% content_for :header_tags do %>
  <meta name="required_script" content="wiki_export_modal" />
<% end %>
```

Reply:
@oliverguenther Thanks for the feedback. How can I access the "this.closeMe()" method. I am bit puzzled here because we are in the regular JS / jQuery context, but I am missing the TS / AngularJS class context. Do you have an idea how to solve that?

Reply:
From a global listener or script, you don't have an immediate angular context so the injector is not there.
The way we expose this is thorugh the OpenProject global: `window.OpenProject.getPluginContext` which returns a promise to a set of services, as well as the injector.

So in the globally dynamic script, you can get the modal Service and close the active modal like so:

```typescript
window
  .OpenProject
  .getPluginContext()
  .then((pluginContext) => {
   const modalService = pluginContext.services.opModalService;
   modalService.close();
});
```